pkg/apis/postgres-operator.crunchydata.com/v1beta1/instrumentation_types.go

```diff
@@ -91,10 +101,70 @@ type InstrumentationLogsSpec struct {
 	RetentionPeriod *Duration `json:"retentionPeriod,omitempty"`
 }
 
+type InstrumentationMetricsSpec struct {
+	// Where users can turn off built-in metrics and also provide their own
+	// custom queries.
+	// ---
+	// +optional
+	CustomQueries *InstrumentationCustomQueriesSpec `json:"customQueries,omitempty"`
+}
+
+type InstrumentationCustomQueriesSpec struct {
+	// User defined queries and metrics.
+	// ---
+	// +optional
+	Add []InstrumentationCustomQueries `json:"add,omitempty"`
+
+	// A list of built-in queries that should be removed. If all queries for a
+	// given SQL statement are removed, the SQL statement will no longer be run.
+	// ---
+	// +optional
+	Remove []string `json:"remove,omitempty"`
+}
+
+type InstrumentationCustomQueries struct {
+	// The name of this batch of queries, which will be used in naming the OTel
+	// SqlQuery receiver.
+	// ---
+	// OTel restricts component names from having whitespace, control characters,
+	// or symbols.
+	// https://github.com/open-telemetry/opentelemetry-collector/blob/main/component/identifiable.go#L23-L26
+	// +kubebuilder:validation:Pattern=`^[^\pZ\pC\pS]+$`
+	//
+	// Set a max length to keep rule costs low.
+	// +kubebuilder:validation:MaxLength=20
+	//
+	// +required
+	Name string `json:"name"`
+
+	// A ConfigMap holding the yaml file that contains the queries.
+	// ---
+	// +required
+	Queries ConfigMapKeyRef `json:"queries"`
+
+	// How often the queries should be run.
+	// ---
+	// Kubernetes ensures the value is in the "duration" format, but go ahead
+	// and loosely validate the format to show some acceptable units.
+	// NOTE: This rejects fractional numbers: https://github.com/kubernetes/kube-openapi/issues/523
+	// +kubebuilder:validation:Pattern=`^((PT)?( *[0-9]+ *(?i:(ms|s|m)|(milli|sec|min)s?))+|0)$`
+	//
+	// `controller-gen` needs to know "Type=string" to allow a "Pattern".
+	// +kubebuilder:validation:Type=string
+	//
+	// Set a max length to keep rule costs low.
+	// +kubebuilder:validation:MaxLength=20
+	// +kubebuilder:validation:XValidation:rule=`duration("0") <= self && self <= duration("60m")`
+	//
+	// +default="5s"
```

Comment:
What happens if you leave this field blank?

Reply:
Tested both having the field with a blank value and deleting the field altogether, and they both result in it defaulting to 5s as expected.